src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcServerTransport.cs

```diff
@@ -17,15 +17,22 @@ internal abstract class IpcServerTransport : IDisposable
         private IIpcServerTransportCallbackInternal _callback;
         private bool _disposed;
 
-        public static IpcServerTransport Create(string transportPath, int maxConnections)
+        public static IpcServerTransport Create(string address, int maxConnections, bool enableTcpIpProtocol)
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            if (!enableTcpIpProtocol || !IpcTcpSocketTransport.TryParseIPAddress(address))
```

Comment:
Was added to make it explicit that you would like to enable TCP support for ReverseDiagnosticServer since it adds additional TCP protocol (and not just namned pipes, unix domain sockets) to the component that in theory could be accessible from all tools that can specify the address used in ReverseDiagnosticServer constructor call. So intent of this flag was a way to explicit enable TCP support in ReverseDiagnosticServer only used by clients explicitly supporting TCP/IP as transport.

Reply:
We (@josalem and @shirhatti) agreed that we should create a separate issue making all tooling smarter to accept TCP/IP for ReversedDiagnosticServer, but initially keep the flag to explicitly disable it unless you instantiate ReversedDiagnosticServer with TCP/IP support. As part of making all tooling smarter to accept TCP/IP as --diagnostic-port we should then drop enableTcpIpProtocol and force usage of tcp:// or similar to explicitly enable TCP/IP from all tools, but until then, we should keep this flag as is.

Reply:
To be clear, this PR doesn't add this functionality to the other tools. This PR only adds the necessary code and the dsrouter tool.

Reply:
Created https://github.com/dotnet/diagnostics/issues/2160 for future work to add functionality to other tools.